core/modules/link/link.module

```diff
@@ -0,0 +1,1460 @@
+<?php
+
+/**
+ * @file
+ * Defines simple link field types.
+ */
+
+define('LINK_EXTERNAL', 'external');
+define('LINK_INTERNAL', 'internal');
+define('LINK_FRONT', 'front');
+define('LINK_EMAIL', 'email');
+define('LINK_NEWS', 'news');
+define('LINK_DOMAINS', 'aero|arpa|asia|biz|build|com|cat|ceo|coop|edu|gov|info|int|jobs|mil|museum|name|nato|net|org|post|pro|tel|travel|mobi|local|xxx');
+
+define('LINK_TARGET_DEFAULT', 'default');
+define('LINK_TARGET_NEW_WINDOW', '_blank');
+define('LINK_TARGET_TOP', '_top');
+define('LINK_TARGET_USER', 'user');
+
+/**
+* Implements hook_config_info().
+*/
+function link_config_info() {
+  $prefixes['link.settings'] = array(
+    'label' => t('Settings'),
+    'group' => t('Link'),
+  );
+  return $prefixes;
+}
+
+/**
+ * Maximum URLs length - needs to match value in link.install.
+ */
+define('LINK_URL_MAX_LENGTH', 2048);
+
+/**
+ * Implements hook_field_info().
+ */
+function link_field_info() {
+  return array(
+    'link_field' => array(
+      'label' => t('Link'),
+      'description' => t('Store a title, href, and attributes in the database to assemble a link.'),
+      'settings' => array(
+        'attributes' => _link_default_attributes(),
+        'url' => 0,
+        'title' => 'optional',
+        'title_value' => '',
+        'title_maxlength' => 128,
+        'enable_tokens' => 1,
+        'display' => array(
+          'url_cutoff' => 80,
+        ),
+      ),
+      'instance_settings' => array(
+        'attributes' => _link_default_attributes(),
+        'url' => 0,
+        'title' => 'optional',
+        'title_value' => '',
+        'title_label_use_field_label' => FALSE,
+        'title_maxlength' => 128,
+        'enable_tokens' => 1,
+        'display' => array(
+          'url_cutoff' => 80,
+        ),
+        'validate_url' => 1,
+        'absolute_url' => 1,
+      ),
+      'default_widget' => 'link_field',
+      'default_formatter' => 'link_default',
+      // Support hook_entity_property_info() from contrib "Entity API".
+      'property_type' => 'field_item_link',
+      'property_callbacks' => array('link_field_property_info_callback'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_instance_settings_form().
+ */
+function link_field_instance_settings_form($field, $instance) {
+  $form = array(
+    '#element_validate' => array('link_field_settings_form_validate'),
+  );
+
+  $form['absolute_url'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Absolute URL'),
+    '#default_value' => isset($instance['settings']['absolute_url']) && ($instance['settings']['absolute_url'] !== '') ? $instance['settings']['absolute_url'] : TRUE,
```

Comment:
All of these `isset($instance['settings']['foo'])` checks in this function should be able to be removed. All defaults are now populated from `hook_field_info()`, both on new and existing fields.